app/src/main/java/com/x8bit/bitwarden/data/vault/datasource/sdk/VaultSdkSourceImpl.kt

```diff
@@ -480,45 +468,36 @@ class VaultSdkSourceImpl(
                     )
 
                 send(result)
-            } catch (e: BitwardenException) {
-                e.asFailure()
-            } finally {
                 close()
+            } catch (e: BitwardenException) {
+                close(e)
```

Comment:
So this just never returns if an error is thrown?

Reply:
`close(e)` propagates the exception. So ultimately, `onFailure` is called in `Fido2CredentialManager`, [here](https://github.com/bitwarden/android/pull/3375/files#diff-d4af48c048c6642a91a4ea480d3ed4b4e1363da7afd0e51318bd42ec867c2541R77).

Reply:
OK, so the `runCatching` handles it?

Reply:
Should we do the same thing in `authenticateFido2Credential`?

Reply:
Indeed!